src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,145 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final RegistrationServiceImpl registrationService =
+            new RegistrationServiceImpl();
+    private static final StorageDao storageDao = new StorageDaoImpl();
+    //Valid users
+    private static final User VALID_USER_ONE = new User();
+    private static final User VALID_USER_TWO = new User();
+    private static final User VALID_USER_THREE = new User();
+    //Valid values
+    //Valid ID's
+    private static final long ID_ONE = 123L;
+    private static final long ID_TWO = 456L;
+    private static final long ID_THREE = 789L;
+    //Valid ages
+    private static final int VALID_AGE_ONE = 18;
+    private static final int VALID_AGE_TWO = 25;
+    private static final int VALID_AGE_THREE = 41;
+    //Valid logins
+    private static final String VALID_LOGIN_ONE = "firstuser@mail.com";
+    private static final String VALID_LOGIN_TWO = "seconduser@mail.com";
+    private static final String VALID_LOGIN_THREE = "thirduser@mail.com";
+    //Valid passwords
+    private static final String VALID_PASSWORD_ONE = "askj435%";
+    private static final String VALID_PASSWORD_TWO = "sdfkjgJKH34";
+    private static final String VALID_PASSWORD_THREE = "skfjgh$53d";
+    //Invalid users
+    private static final User EMPTY_USER = new User();
+    private static final User CUSTOM_USER = new User();
+    //Invalid values
+    //Invalid ages
+    private static final int LOW_AGE_ONE = 0;
+    private static final int LOW_AGE_TWO = 15;
+    private static final Integer[] INVALID_AGE_ARRAY = {Integer.valueOf(LOW_AGE_ONE),
+            Integer.valueOf(LOW_AGE_TWO),
+            null};
+    //Invalid logins
+    private static final String LOGIN_WITH_WRONG_FIRST = "%sdjfh@mail.com";
+    private static final String SHORT_LOGIN_ONE = "";
+    private static final String SHORT_LOGIN_TWO = "short";
+    private static final String EXISTING_LOGIN = "firstuser@mail.com";
+    private static final String[] INVALID_LOGIN_ARRAY = {LOGIN_WITH_WRONG_FIRST,
+            SHORT_LOGIN_ONE,
+            SHORT_LOGIN_TWO,
+            EXISTING_LOGIN,
+            null};
+    //Invalid passwords
+    private static final String SHORT_PASSWORD_ONE = "";
+    private static final String SHORT_PASSWORD_TWO = "123";
+    private static final String SHORT_PASSWORD_THREE = "12345";
+    private static final String[] INVALID_PASSWORD_ARRAY = {SHORT_PASSWORD_ONE,
+            SHORT_PASSWORD_TWO,
+            SHORT_PASSWORD_THREE,
+            null};
```

Comment:
Instead of a lot of constants you can create users with valid and invalid data, use private methods for this users